packages/protocol/contracts/L1/TaikoData.sol

```diff
@@ -17,10 +17,11 @@ library TaikoData {
         // This number is calculated from blockMaxProposals to make
         // the 'the maximum value of the multiplier' close to 20.0
         uint256 blockMaxVerificationsPerTx;
-        uint32 blockMaxGasLimit;
+        // The max gas used per block, including anchor.
+        uint32 blockAndTxMaxGasUsed;
```

Comment:
Just want to double check if we will use this value to reject transactions with `tx.limit > blockAndTxMaxGasUsed` because then the name is a bit confusing (gasLimit vs gasUsed). I'm not sure if Ethereum itself already does something like this e.g. if a tx has a gas limit of 100M gas but the block gas limit is 10M gas, is it rejected or can it still be included in a block if the tx ends up only using < 10M gas?

Reply:
<img width="362" alt="image" src="https://github.com/taikoxyz/taiko-mono/assets/104078303/692b12b4-e162-4c2d-8632-4020c9205952">

No all transactions accumulated gasLimit can not exceed `block.gasLimit`, so in go-ethereum, if its building a block, that transaction will just be rejected; if it received a block like this from the network, the whole block will be treat as invalid. So i think, if we just use `blockAndTxMaxGasUsed` as `block.gasLimit`, it will break the ethereum protocol, maybe use `uint256.max` or just set a `txMaxGasLimit` then use the all txs accumulated gasLimit as the `block.gasLimit` will be more ethereum competitive?

Reply:
So Ethereum has a pretty similar problem we currently have? As in only a single tx with a tx.gasLimit close to the block gas limit can be included? The only difference with our previous approach is that smaller transactions cannot be included anymore while in Ethereum blocks that would still be possible.

I think `blockAndTxMaxGasUsed` should still be okay no? So there is a block gas limit, and each `tx.gasLimit` needs to be  `<= block.gasLimit`. But to be 100% compatible with Ethereum we would also have to add an additional check that `l(Br)_u + tx.gasLimit <= block.gasLimit` as well. That seems like a pretty easy check to add.

Reply:
> set a txMaxGasLimit then use the all txs accumulated gasLimit as the block.gasLimit will be more ethereum competitive?

I think this would be a pretty big change (similar to the previous approach) where the the `tx.gasLimit` is a pretty important value. In Ethereum it's a mostly unimportant value where you can set it to a high value to make sure it will be successful with no real impact (except I guess for the rule above, but for any real use case should still be much smaller than block.gasLimit).